File: appium-framework/pages/HistoryPage.ts
import BasePage from './BasePage';

export default class HistoryPage extends BasePage {
    get headerTitle() { return '//*[@text="Interview History"]'; }
    get noHistoryText() { return '//*[@text="No interview history found"]'; }
    get historyItem() { return '//*[contains(@text, "Mock Interview") or contains(@text, "%")]'; }
    get deleteButton() { return '//*[@resource-id="btn-delete-log-record" or contains(@content-desc, "delete") or contains(@text, "Delete")]'; }

    async hasHistoryItems(): Promise<boolean> {
        return await this.isDisplayed(this.historyItem);
    }

    async clickHistoryItem() {
        await this.click(this.historyItem);
    }

    async deleteHistoryItem() {
        if (await this.isDisplayed(this.deleteButton)) {
            await this.click(this.deleteButton);
        }
    }
}

File: tests/page_objects/history_page.py
from selenium.webdriver.common.by import By
from tests.page_objects.base_page import BasePage

class HistoryPage(BasePage):
    HISTORY_ROWS = (By.CLASS_NAME, "history-item-row")
    DELETE_BUTTON = (By.ID, "btn-delete-log-record")

    def get_history_rows(self):
        return self.find_elements(self.HISTORY_ROWS)

    def delete_history_item(self):
        self.click(self.DELETE_BUTTON)

    def is_delete_button_displayed(self):
        return self.is_displayed(self.DELETE_BUTTON)
